src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -399,6 +399,72 @@ inline std::istream& operator>>(std::istream& is, SchedulingCoreType& core_type)
+    PIPELINE_PARTITION = 2,  // Split one model into parts and run each socket/device in parallel as a pipeline.
+    ENTIRE_PLATFORM = 3,     // Run one model on the entire platform with all sockets/devices.
+    SINGLE_DEVICE = 4,       // Run one model on single socket/device.
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const LlmDistributionPolicy& stream_mode) {
+    switch (stream_mode) {
+    case LlmDistributionPolicy::TENSOR_PARTITION:
+        return os << "TENSOR_PARTITION";
+    case LlmDistributionPolicy::DATA_PARTITION:
+        return os << "DATA_PARTITION";
+    case LlmDistributionPolicy::PIPELINE_PARTITION:
+        return os << "PIPELINE_PARTITION";
+    case LlmDistributionPolicy::ENTIRE_PLATFORM:
+        return os << "ENTIRE_PLATFORM";
+    case LlmDistributionPolicy::SINGLE_DEVICE:
+        return os << "SINGLE_DEVICE";
+    default:
+        OPENVINO_THROW("Unsupported LLM distribution policy!");
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, LlmDistributionPolicy& stream_mode) {
+    std::string str;
+    is >> str;
+    if (str == "TENSOR_PARTITION") {
+        stream_mode = LlmDistributionPolicy::TENSOR_PARTITION;
+    } else if (str == "DATA_PARTITION") {
+        stream_mode = LlmDistributionPolicy::DATA_PARTITION;
+    } else if (str == "PIPELINE_PARTITION") {
+        stream_mode = LlmDistributionPolicy::PIPELINE_PARTITION;
+    } else if (str == "ENTIRE_PLATFORM") {
+        stream_mode = LlmDistributionPolicy::ENTIRE_PLATFORM;
+    } else if (str == "SINGLE_DEVICE") {
+        stream_mode = LlmDistributionPolicy::SINGLE_DEVICE;
+    } else {
+        OPENVINO_THROW("Unsupported LLM distribution policy: ", str);
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief This property defines distribution policy for Large language models (LLM).
+ * @ingroup ov_runtime_cpp_prop_api
+ *
+ * Developer can use this property to select LLM distribution policy for CPU inference with multiple sockets platform or
+ * GPU inference with multiple GPU devices.
+ * -- TENSOR_PARTITION   : Split one node or subgraph into parts and run one part per socket/device in parallel.
+ * -- DATA_PARTITION     : Split one batch input into parts and run one part per socket/device in parallel.
+ * -- PIPELINE_PARTITION : Split one model into parts and run each socket/device in parallel as a pipeline.
+ * -- ENTIRE_PLATFORM    : Run one model on the entire platform with all sockets/devices.
+ * -- SINGLE_DEVICE      : Run one model on single socket/device.
+ *
+ * The following code is an example to only use all threads of one socket for one stream on dual sockets platform.
+ *
+ * @code
+ * ie.set_property(ov::hint::llm_distribution_policy(ov::hint::LlmDistributionPolicy::SINGLE_DEVICE));
+ * @endcode
+ */
+static constexpr Property<LlmDistributionPolicy> llm_distribution_policy{"LLM_DISTRIBUTION_POLICY"};
```

Comment:
Based on offline discussions:
1. Lets rename to: `ov::hint::distribution_policy`
2. Available set of options: TENSOR_PARALLEL, DATA_PARALLEL, PIPELINE_PARALLEL 
3. Property itslef should allow to combine diffrent policies (e.g. TENSOR_PARALLEL + DATA_PARALLEL). So it is reasonable to use std::unordered_set (or just std::set) with DistributionPolicy type. Empty set corresponds will represent disabled parallellism

@AlexKoff88 @yury-gorbachev @ilya-lavrenov any comments?

Reply:
Updated. 

@yury-gorbachev @dmitry-gorokhov @ilya-lavrenov @AlexKoff88 Please review and comments.